apps/emqx/src/emqx_authentication.erl

```diff
@@ -617,18 +629,17 @@ maybe_unhook(#{hooked := true} = State) ->
 maybe_unhook(State) ->
     State.
 
-do_create_authenticator(ChainName, AuthenticatorID, #{enable := Enable} = Config, Providers) ->
+do_create_authenticator(AuthenticatorID, #{enable := Enable} = Config, Providers) ->
     case maps:get(authn_type(Config), Providers, undefined) of
         undefined ->
             {error, no_available_provider};
         Provider ->
-            Unique = unique(ChainName, AuthenticatorID, ?VER_1),
-            case Provider:create(Config#{'_unique' => Unique}) of
+            case Provider:create(Config) of
```

Comment:
As far as I can understand, `version` is used only to generate resource ids, so that we can:
* create new resource with new id while the old is still active;
* avoid resource leakage.

I moved `resource_id` handling to the providers and added cleanup to the authentication initialization process.